interactive-etl/README.md

```diff
@@ -0,0 +1,330 @@
+
+First we define the table, the column types and the connection configuration. For `upsert-kafka` based tables you need to provide the `PRIMARY_KEY` as this will be used as the key for the message (each new row) sent to Kafka:
+
+```sql
+CREATE TABLE CleanedSalesRecordTable ( 
+    invoice_id STRING, 
+    user_id STRING, 
+    product_id STRING, 
+    quantity INT, 
+    unit_cost_gbp INT, 
+    purchase_time TIMESTAMP(3),
+    PRIMARY KEY (`user_id`) NOT ENFORCED
+) WITH ( 
+    'connector' = 'upsert-kafka', 
+    'topic' = 'flink.cleaned.sales.records.interactive', 
+    'properties.bootstrap.servers' = 'my-cluster-kafka-bootstrap.flink.svc:9092', 
+    'properties.client.id' = 'sql-cleaning-client', 
+    'properties.transaction.timeout.ms' = '800000', 
+    'key.format' = 'csv', 
+    'value.format' = 'csv', 
+    'value.fields-include' = 'ALL' 
+);
+```
+
+For the `upsert-kafka` connector you are providing the Kafka producer configuration and so you need to tell the connector what format the keys and values will be in. 
+If you wanted to use Avro via Apicurio (like the source table we used above) you would need to first define the schema in Apicurio Registry and then reference that in the connector configuration.
+For simplicity, we are using a simple CSV format for the key and the values.
+
+Now you have the output table defined you can insert the results of the formatting query into it:
+
+```sql
+INSERT INTO CleanedSalesRecordTable
+SELECT 
+    invoice_id, 
+    user_id,
+    product_id,
+    CAST(quantity AS INT), 
+    CAST(TRIM(LEADING '£' from unit_cost) AS INT), 
+    purchase_time
+FROM SalesRecordTable;
+```
+
+Issuing this query will submit a Flink streaming Job to the session cluster.
+
+```log
+[INFO] Submitting SQL update statement to the cluster...
+[INFO] SQL update statement has been successfully submitted to the cluster:
+Job ID: 2a8c1677c64854482f22ca96295eec65
+```
+
+However, this doesn't mean everything is working correctly.
+This just means you have successfully sent the jar containing the query code to the Flink cluster.
+You can check if the job is actually running using `SHOW JOBS;`:
+
+```
++----------------------------------+-------------------------+----------+-------------------------+
+|                           job id |                job name |   status |              start time |
++----------------------------------+-------------------------+----------+-------------------------+
+| 198e2bb14fffda57b303179d994d9ac1 |                 collect | CANCELED | 2024-10-04T13:21:30.660 |
+| 3030185a27ea5a826d1cabef84e70b41 |                 collect | CANCELED | 2024-10-04T13:18:02.888 |
+| 2a8c1677c64854482f22ca96295eec65 | CleanedSalesRecordTable |  RUNNING | 2024-10-04T14:02:45.643 |
++----------------------------------+-------------------------+----------+-------------------------+
+```
+
+Or you can look at the Flink UI (which you port-forwarded above) on `localhost:8081`.
```

Comment:
I know this only targets minikube but is it wort calling out this URL only works there?

Reply:
As @showuon pointed out the UI isn't support yet so I will just remove it for now

Reply:
Yeah I was considering making the same observation but didn't as that's a downstream decision not upstream and this is upstream.

That said happy to skip the UI as well